### DISCUSSION TITLE
Do not run tests on PR merge to master

### DISCUSSION THREAD

**Discussion by dschinkel** [community/community#26841] [Actions]
<p>Is there a way to tell actions not to run my tests.yml file on a PR merge to master?  I only care about running tests:</p>
<ul>
<li>When initial PR is Created</li>
<li>subsequent commits to an open PR</li>
</ul>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/dschinkel">@dschinkel</a> ,</p>
<p>Once a PR is merged, it will be closed automatically. The activity type ‘<strong>closed</strong>’ of the <strong>pull_request</strong> event should work.</p>
<p>In your case, you can reference to the example workflow below to configure yours:</p>
<pre><code class="lang-yaml">on:
  pull_request:
    types: [opened, synchronize, closed]
    branches:
      - master

jobs:
  build:
  # Some jobs do not have any if statement.
  # They will always be executed in the workflow. For example, 'build' job.

  test:
    if: ${{github.event_name == 'pull_request' &amp;&amp; github.event.action != 'closed'}}
  # With this if statement, the 'test' job will only run
  # when the activity type of the PR is not closed.
</code></pre>
<p>Of course, you also can remove ‘<strong>closed</strong>’ from the list of the activity types, so that the workflow will not be triggered every time the PR is closed (manual or merged).</p>
<pre><code class="lang-yaml">on:
  pull_request:
    types: [opened, synchronize]
    branches:
      - master
</code></pre>

**Comment by dschinkel** [comment]
<p>Thanks.  I also had to remove synchronized as well.  Now it only runs my CI on PR merges (closed) which is what I’m aiming for.</p>
